controllers/clusterversion_controller.go

```diff
@@ -139,6 +141,32 @@ func (c *ClusterVersionReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
+	// deletion phase
+	if !instance.GetDeletionTimestamp().IsZero() {
+		result, err := c.deletionPhase()
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		if !result.Requeue {
+			//remove finalizer
+			controllerutil.RemoveFinalizer(&instance, ClusterVersionFinalizer)
+			if err := c.Client.Update(c.ctx, &instance); err != nil {
+				return ctrl.Result{}, err
+			}
+			c.log.Info("finallizer removed successfully")
+		}
+		return result, nil
+	}
+
+	//ensure finalizer
+	if !controllerutil.ContainsFinalizer(&instance, ClusterVersionFinalizer) {
+		c.log.Info("finalizer missing on the ClusterVersion resource, adding...")
+		controllerutil.AddFinalizer(&instance, ClusterVersionFinalizer)
+		if err := c.Client.Update(c.ctx, &instance); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
```

Comment:
we'll mostly not be around when we a deletion event is triggered on `ClusterVersion` basically, it's a cluster teardown.

Reply:
discussed offline to split the changes originally suggested under #64 to several PRs in order to better incapsulate a single change in each of them, that is correct that we will not be around when the cluster is tearing down but this change will also not break our controller (still needs to be tested) after PR #64 will take into affect this code should as well. if you think we should base it on one another that is an option as well but i thought this approach will allow us to get them both in since they are not directly related.